core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationEditorDirective.js

```diff
@@ -0,0 +1,149 @@
+              hide_complex_extensions: 'true'
+            }
+          };
+          var showMarkAllAudioAsNeedingUpdateModalIfRequired = function(
+              contentId) {
+            var stateName = StateEditorService.getActiveStateName();
+            var state = ExplorationStatesService.getState(stateName);
+            var contentIdsToAudioTranslations = (
+              state.contentIdsToAudioTranslations);
+            if (contentIdsToAudioTranslations.hasUnflaggedAudioTranslations(
+              contentId)) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/components/forms/' +
+                  'mark_all_audio_as_needing_update_modal_directive.html'),
+                backdrop: true,
+                resolve: {
+                  considerTranslation: function() {
+                    return false;
+                  }
+                },
+                controller: 'MarkAllAudioAsNeedingUpdateController'
+              }).result.then(function() {
+                contentIdsToAudioTranslations.markAllAudioAsNeedingUpdate(
+                  contentId);
+                ExplorationStatesService.saveContentIdsToAudioTranslations(
+                  stateName, contentIdsToAudioTranslations);
+              });
+            }
+          };
+          var contentId = null;
+          var langaugeCode = null;
+          $scope.activeWrittenTranslation = null;
+          $scope.translationEditorIsOpen = false;
+          $scope.isTranslatable = EditabilityService.isTranslatable;
+          $scope.$on('activeContentIdChanged', function() {
+            initEditor();
+          });
+          $scope.$on('activeLanguageChanged', function() {
+            initEditor();
+          });
+
+          var initEditor = function() {
+            $scope.activeWrittenTranslation = null;
+            $scope.translationEditorIsOpen = false;
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService.displayed.hasWrittenTranslation(
+              contentId, langaugeCode)) {
+              $scope.activeWrittenTranslation = (
+                StateWrittenTranslationsService.displayed
+                  .getWrittenTranslation(contentId, langaugeCode));
+            }
+          };
+          initEditor();
+          var saveTranslation = function() {
+            var oldTranslation = '';
+            var newTranslation = '';
+            contentId = (
+              TranslationTabActiveContentIdService.getActiveContentId());
+            langaugeCode = TranslationLanguageService.getActiveLanguageCode();
+            if (StateWrittenTranslationsService
+              .savedMemento.hasWrittenTranslation(contentId, langaugeCode)) {
+              oldTranslation = (StateWrittenTranslationsService
```

Comment:
Why was the variable initialized to an empty string? I think it is getting assigned to a writtenTranslation object right?